src/buildings/nobMilitary.cpp

```diff
@@ -329,60 +329,47 @@ void nobMilitary::LookForEnemyBuildings(const nobBaseMilitary* const exception)
         if(*it != exception && (*it)->GetPlayer() != player && gwg->GetPlayer((*it)->GetPlayer()).IsAttackable(player))
         {
             unsigned distance = gwg->CalcDistance(pos, (*it)->GetPos());
+            FrontierDistance newFrontierDistance = DIST_FAR;
 
-            // check if military building is reachable
-            if(gwg->GetGGS().isEnabled(AddonId::FRONTIER_DISTANCE_REACHABLE)
-               && !DoesReachablePathExist(*gwg, (*it)->GetPos(), pos, MAX_ATTACKING_RUN_DISTANCE))
-            {
-                frontier_distance = DIST_FAR;
-            }
-            // in nahem Umkreis, also Grenzen berühren sich
-            else if(distance <= MILITARY_RADIUS[size] + (*it)->GetMilitaryRadius()) // warum erzeugtn das ne warning in vs2008?
+            if(distance <= MILITARY_RADIUS[size] + (*it)->GetMilitaryRadius())
             {
-                // Grenznähe entsprechend setzen
-                frontier_distance = DIST_NEAR;
-
-                // Wenns ein richtiges Militärgebäude ist, dann dort auch entsprechend setzen
-                if(BuildingProperties::IsMilitary((*it)->GetBuildingType()))
-                    static_cast<nobMilitary*>(*it)->NewEnemyMilitaryBuilding(3);
+                newFrontierDistance = DIST_NEAR;
             }
             // in mittlerem Umkreis, also theoretisch angreifbar?
             else if(distance < BASE_ATTACKING_DISTANCE + (GetMaxTroopsCt() - 1) * EXTENDED_ATTACKING_DISTANCE)
             {
-                // Grenznähe entsprechend setzen
-                if(!frontier_distance)
-                    frontier_distance = DIST_MID;
-
-                // Wenns ein richtiges Militärgebäude ist, dann dort auch entsprechend setzen
-                if(BuildingProperties::IsMilitary((*it)->GetBuildingType()))
-                    static_cast<nobMilitary*>(*it)->NewEnemyMilitaryBuilding(1);
-            }
-            // andere Richtung muss auch getestet werden, zumindest wenns eine normaler Militärgebäude ist, Bug 389843
-            else if((*it)->GetGOT() == GOT_NOB_MILITARY)
+                newFrontierDistance = DIST_MID;
+            } else if((*it)->GetGOT() == GOT_NOB_MILITARY)
             {
                 nobMilitary* mil = dynamic_cast<nobMilitary*>(*it);
```

Comment:
Use safe_cast here to improve further (dynamic_cast in debug, static_cast in release mode)

Reply:
as i am no c++ professional, i don't now safe_cast as cast operation. after googling around i found only a Microsoft command extension which is named safe_cast.
i could do a static_cast if you want.

Reply:
It's something I introduced. Maybe I should rename it to something more common (some C++ helps use `down_cast` etc). The idea is the same: Use static_cast/dynamic_cast in a slim wrapper to capture the intent.

My version is `checkedCast` (https://github.com/Return-To-The-Roots/s25client/blob/master/common/include/commonDefines.h#L137) but I should enhance it... However it is to be used the same as `static_cast`: `safeCast<nobMilitary*>(src)` or similar.